### ISSUE TITLE
can't use use vim.keymap.set to set mapping

### ISSUE THREAD

**Issue by jam1015** [leath-dub/snipe.nvim#23]
reprex:

```lua

local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not (vim.uv or vim.loop).fs_stat(lazypath) then
  local lazyrepo = "https://github.com/folke/lazy.nvim.git"
  local out = vim.fn.system({ "git", "clone", "--filter=blob:none", "--branch=stable", lazyrepo, lazypath })
  if vim.v.shell_error ~= 0 then
    vim.api.nvim_echo({
      { "Failed to clone lazy.nvim:\n", "ErrorMsg" },
      { out,                            "WarningMsg" },
      { "\nPress any key to exit..." },
    }, true, {})
    vim.fn.getchar()
    os.exit(1)
  end
end
vim.opt.rtp:prepend(lazypath)
vim.g.mapleader = " "
vim.g.maplocalleader = "\\"
require("lazy").setup({
  spec = {
  {
    "leath-dub/snipe.nvim",
    --keys = {
     -- { "<leader>bb", function() require("snipe").open_buffer_menu() end, desc = "Open Snipe buffer menu" }
    --},
    opts = require("plugin_configs.snipe_opts"),
    config = function()
      vim.keymap.set("n", "<leader>bb", function () require("snipe").open_buffer_menu() end)
    end
  },
  
  },
})

```

when I run `<leader>bb` I get:

```
E5108: Error executing lua: ...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:417: bad argument #1 to 'rep' (string expected, got nil)               
stack traceback:                                                                                                                                               
        [C]: in function 'rep'                                                                                                                                 
        ...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:417: in function 'generate_tags'                                                           
        ...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:360: in function 'annotate_with_tags'                                                      
        ...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:116: in function 'open'                                                                    
        ...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:214: in function <...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:210>         
        ...dan/.local/share/nvim/lazy/snipe.nvim/lua/snipe/init.lua:233: in function 'open_buffer_menu'                                                        
        /home/jordan/dotfiles/.config/nvim/init.lua:40: in function </home/jordan/dotfiles/.config/nvim/init.lua:40>                                           
Press ENTER or type command to continue    
```

**Comment by leath-dub** [comment]
Can I see the options you have set ?

**Comment by leath-dub** [comment]
For now you can avoid that specific function ( it is a point of contention  and why I want to refactor ):

```lua
  vim.keymap.set("n", "<leader>bb", require("snipe").create_buffer_menu_toggler())
```

**Comment by jam1015** [comment]
> Can I see the options you have set ?

thanks for the advice! just running with the reprex should be enough to cause the problem

**Comment by leath-dub** [comment]
> > Can I see the options you have set ?
> 
> thanks for the advice! just running with the reprex should be enough to cause the problem

Ok I got a chance to look at this, it works if you change `config` to `init` I am not really sure why but some state of the module is getting overwritten for some odd reason.

**Comment by leath-dub** [comment]
TLDR; `setup` is not being called.

ahh I figured it out I think the issue is actually that lazy is ignoring the `opts` field and not calling setup because you are also passing a `config` field. You can just have this 
instead:

```lua
  {
    "leath-dub/snipe.nvim",
    --keys = {
     -- { "<leader>bb", function() require("snipe").open_buffer_menu() end, desc = "Open Snipe buffer menu" }
    --},
    config = function()
      require("snipe").setup()
      vim.keymap.set("n", "<leader>bb", function () require("snipe").open_buffer_menu() end)
    end
  },
```
